test_scripts/API/Additional_Submenus/002_legacy_app_parentID_filtered.lua

```diff
@@ -0,0 +1,65 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal:
+-- https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0148-template-additional-submenus.md#backwards-compatibility
+-- Description: Tests sending a parentID param in a AddSubMenu request with a legacy RPC spec version
+-- In case:
+-- 1) Mobile application is set to appropriate HMI level and System Context MENU, MAIN
+-- 2) Mobile application sends AddSubMenu SubMenu with menuID = 1
+-- 3) Mobile sends additional AddSubMenu request with menuID = 99 and parentID = 1
+-- SDL does:
+-- 1) Sends AddSubMenu requests to HMI with no parentID included in the requests
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Smoke/commonSmoke')
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+config.application1.registerAppInterfaceParams.syncMsgVersion = {
+    majorVersion = 6,
+    minorVersion = 0
+}
+
+--[[ Local Variables ]]
+local requestParams = {
+    menuID = 99, 
+    menuName = "SubMenu2",
+    parentID = 1
+}
+
+local hmiRequestParams = {
+    menuID = 99, 
+    menuParams = { 
+        menuName = "SubMenu2"
+    }
+}
+ 
+local function AdditionalSubmenu()
+    local cid = common.getMobileSession():SendRPC("AddSubMenu", requestParams)
+    common.getHMIConnection():ExpectRequest("UI.AddSubMenu", hmiRequestParams)
+    :ValidIf(function(_, data)
+        return data.params.menuParams["parentID"] == nil
+      end)
+    :Do(function(_, data)
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+      end)
+    common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+    common.getMobileSession():ExpectNotification("OnHashChange")
+    :Do(function(_, data)
+        common.hashId = data.payload.hashID
+      end)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("App registration", common.registerApp)
+
+runner.Title("Test")
+runner.Step("App activate, HMI SystemContext MAIN", common.activateApp)
+runner.Step("Add menu", common.addSubMenu)
+runner.Step("Add additional submenu for legacy app", AdditionalSubmenu)
+
+runner.Title("Postconditions")
+runner.Step("Stop SDL", common.postconditions)
```

Comment:
Should include duplicate name cases as well. One with two submenus on the same level (`DUPLICATE_NAME` response) and another with submenus on separate levels (`SUCCESS` response)

Reply:
Might also be a good place to add more cases for AddSubMenu/AddCommand/DeleteSubMenu/DeleteCommand. Would likely be only if time allows though.

Reply:
One `DeleteSubMenu` test that definitely should be covered is deleting a submenu which contains both nested submenus and commands.